libs/vkd3d/resource.c

```diff
@@ -522,12 +524,22 @@ static HRESULT vkd3d_get_image_create_info(struct d3d12_device *device,
     image_info->sType = VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO;
     image_info->pNext = NULL;
     image_info->flags = 0;
+
+    if (resource && (resource->heap_flags & D3D12_HEAP_FLAG_SHARED))
```

Comment:
Should validate here that we can only deal with simple 2D textures, otherwise just fail the call with E_NOTIMPL.

Reply:
I decided to instead put the logic checking this in ::CreateSharedHandle where we set the metadata, as for all we know it could be valid to shared non 2D textures within D3D12 and we just need an updated metadata structure.